lib/rubygems/commands/push_command.rb

```diff
@@ -98,19 +89,7 @@ def send_gem(name)
       terminate_interaction 1
     end
 
-    gem_data = Gem::Package.new(name)
-
-    unless @host
-      @host = gem_data.spec.metadata['default_gem_server']
-    end
-
-    push_host = nil
-
-    if gem_data.spec.metadata.has_key?('allowed_push_host')
-      push_host = gem_data.spec.metadata['allowed_push_host']
-    end
-
-    @host ||= push_host
```

Comment:
host will not be nil if `push_host` is present so assignment is not useful, use the same mechanism as in `execute` to standardise access to gem metadata